dictdatabase/index_manager.py

```diff
@@ -0,0 +1,28 @@
+import hashlib
+
+from dictdatabase import utils
+
+
+class IndexManager:
+    @staticmethod
+    def create_index(all_file_bytes: bytes, key: str, start, end):
+        """
+        It takes a JSON file, a key, and a start and end position, and returns a tuple of information about the key and its
+        value
+
+        Args:
+          all_file_bytes (bytes): The entire file as a byte string.
+          key (str): The key of the value we're indexing.
+          start: the start of the value in the file
+          end: the end of the value in the file
+
+        Returns:
+          The key, start, end, indent_level, indent_with, value_hash, end
+        """
+        key_start, key_end = utils.find_outermost_key_in_json_bytes(all_file_bytes, key)
+        indent_level, indent_with = utils.detect_indentation_in_json_bytes(
+            all_file_bytes, key_start
+        )
+        value_bytes = all_file_bytes[start:end]
+        value_hash = hashlib.sha256(value_bytes).hexdigest()
+        return key, start, end, indent_level, indent_with, value_hash, end
```

Comment:
Maybe return a new class here that is a dataclass named "IndexRecord"

Reply:
Yep, planned to do this in the next commits

Reply:
Okay! By the way, you can run `./cli.sh -bp` to run multiprocessing benchmarks, and see how performance is affected by using the IndexRecord

Reply:
Oh yeah, thats a folder that I can't share because it is out of an production project, but I was using it because it is a good example for a large file